clickhouse-native-jdbc/src/main/java/com/github/housepower/jdbc/ClickhouseJdbcUrlParser.java

```diff
@@ -32,92 +34,57 @@
     public static final String CLICKHOUSE_PREFIX = "clickhouse:";
     public static final String JDBC_CLICKHOUSE_PREFIX = JDBC_PREFIX + CLICKHOUSE_PREFIX;
 
-    public static final Pattern DB_PATH_PATTERN = Pattern.compile("/([a-zA-Z0-9_]+)");
-    public static final Pattern HOST_PORT_PATH_PATTERN = Pattern.compile("//(?<host>[^/:\\s]+)(:(?<port>\\d+))?");
+    public static final String HOST_DELIMITER = ",";
+    public static final String PORT_DELIMITER = ":";
+
+    public static final Pattern CONNECTION_PATTERN = Pattern.compile("//(?<hosts>[^/?\\s]+)(?:/(?<database>([a-zA-Z0-9_]+)))?(?:\\?(?<query>.*))?");
```

Comment:
How does `ck1,ck2:1234` be interpreted? 
1) `ck1:default_port` and `ck2:1234` or 
2) `ck1:1234` and `ck2:1234`

Reply:
Any behavior look reasonable, please just add clear comments here.

Reply:
The `query` is confusing, we can learn something from MySQL[1] or Apache Hive[2], maybe `vars` or `properties` is a proper name.

[1] https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-jdbc-url-format.html
[2] https://cwiki.apache.org/confluence/display/hive/hiveserver2+clients#HiveServer2Clients-ConnectionURLs

Reply:
Some bad but recoverable pattern `ck1:`, `,ck1`, `ck1,ck2,`, `ck1:,ck2`

Reply:
Another bad but recoverable patterns `jdbc:clickhouse://ck1,ck2/`

Reply:
already fix